### PR TITLE
[335] Beta registration

### PR DESCRIPTION
**Pull request by AndrewScibek**

### CODE REVIEW

**Comment by AndrewScibek** [comment]
[Create beta registration](https://trello.com/c/gP4giiC6/335-create-beta-registration)

File: app/services/authentication.py [hunk 1]
```diff
@@ -13,7 +17,7 @@ def decode_token(token, expiration=3600):
     try:
         email = serializer.loads(token, salt="email-confirm-salt", max_age=expiration)
         return email
-    except Exception as e:
+    except Exception:
```
Comment (stephenprater) [inline app/services/authentication.py]: Why are we capturing `Exception` here.  Is there nothing higher up we can catch?

File: app/routes/authentication.py [hunk 2]
```diff
@@ -84,15 +84,22 @@ def register_user():
...
     try:
-        newuser.set_password(password)
-        db.session.add(newuser)
+        new_user.set_password(password)
+        db.session.add(new_user)
         db.session.flush()
-        authentication.send_registration_email(email)
     except IntegrityError:
         return json_api(UnprocessableEntity, ErrorSchema), 422
+    if reg_user_count > 1000:
```
Comment (stephenprater) [inline app/routes/authentication.py]: This feels like the wrong place for this logic.

I think we should consider something like

`authentication.send_registration_email(user.id)`

And then have _that_ method determine what email to send.

File: app/jobs/email.py [hunk 3]
```diff
@@ -6,13 +6,51 @@
...
+        self.smtp_password = current_app.config["MAIL_PASSWORD"]
+        self.host = current_app.config["MAIL_SERVER"]
+        self.port = current_app.config["MAIL_PORT"]
+        self.tls = current_app.config["MAIL_USE_TLS"]
+
+
+@Q.job(timeout=60000)
+def send_beta_backlog_notification_email(email_address):
+    config = EmailConfig()
+    msg = MIMEMultipart("alternative")
```
Comment (stephenprater) [inline app/jobs/email.py]: What if we moved all of this configuration into Email classes and then gave those classes the responsibility to send the correct email.   That would let us move "Emails" into a different object hierarchy so we're not storing it all inside the body of these job functions.

**Pull request merged**